File: tests/fixtures/sample.py
"""Fibonacci and prime fixtures for tree-sitter parser tests."""
import math


def fibonacci(n, memo=None):
    """Return the nth Fibonacci number (memoised)."""
    if memo is None:
        memo = {}
    if n in memo:
        return memo[n]
    if n <= 1:
        return n
    memo[n] = fibonacci(n - 1, memo) + fibonacci(n - 2, memo)
    return memo[n]


def is_prime(n):
    """Return True if n is a prime number."""
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True


primes_under_50 = [x for x in range(2, 50) if is_prime(x)]

if __name__ == "__main__":
    print([fibonacci(i) for i in range(10)])
    print(primes_under_50)

File: tests/fixtures/sample.rs
// Fibonacci and prime fixtures for tree-sitter parser tests.

fn fibonacci(n: u64) -> u64 {
    match n {
        0 => 0,
        1 => 1,
        _ => fibonacci(n - 1) + fibonacci(n - 2),
    }
}

fn is_prime(n: u64) -> bool {
    if n < 2 {
        return false;
    }
    let limit = (n as f64).sqrt() as u64 + 1;
    for i in 2..limit {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    let fibs: Vec<u64> = (0..10).map(fibonacci).collect();
    println!("{:?}", fibs);

    let primes: Vec<u64> = (2..50).filter(|&x| is_prime(x)).collect();
    println!("{:?}", primes);
}
